crates/codegen/parser/runtime/src/support/parser_function.rs

```diff
@@ -17,11 +18,35 @@ where
 
 impl<L, F> ParserFunction<L> for F
 where
+    L: Lexer,
     F: Fn(&L, &mut ParserContext<'_>) -> ParserResult,
 {
     fn parse(&self, language: &L, input: &str) -> ParseOutput {
         let mut stream = ParserContext::new(input);
-        let result = self(language, &mut stream);
+        let mut result = self(language, &mut stream);
+
+        // For a succesful/recovered parse, collect any remaining trivia as part of the parse result
+        if let ParserResult::Match(r#match) = &mut result {
+            let [topmost] = r#match.nodes.as_mut_slice() else {
+                unreachable!(
+                    "Match at the top level of a parse does not have exactly one Rule node"
+                )
+            };
+
+            let eof_trivia = match Lexer::leading_trivia(language, &mut stream) {
+                ParserResult::Match(eof_trivia) if !eof_trivia.nodes.is_empty() => {
```

Comment:
Q: What happens in the case there were errors in parsing these trivia? like for example:

```solidity
contract X {}

/* unterminated comment
```

Reply:
The trivia is not matched, so it's not appended; in which case, we later SKIP over it as part of the unexpected sequence. I've added a test in https://github.com/NomicFoundation/slang/pull/733/commits/5973ece5afffa01c45ceb6eeba8837e7eee59ad6

Reply:
Thanks